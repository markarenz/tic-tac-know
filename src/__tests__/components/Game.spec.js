import React from 'react';
import { mockRandom } from 'jest-mock-random';
import { render, fireEvent, screen, waitFor } from '@testing-library/react';
import Game from '../../components/Game';
import { AI_LEVELS, X } from '../../constants';
import { act } from 'react-dom/test-utils';

beforeEach(() => {
  mockRandom(0);
});

describe('Game', () => {
  const mocks = {
    aiLevel: AI_LEVELS.AI_LEVEL_RND,
    goToMenu: jest.fn(),
    playerSide: X,
  };
  it('renders game', () => {
    const container = render(
      <Game aiLevel={mocks.aiLevel} goToMenu={mocks.goToMenu} playerSide={mocks.playerSide} />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders result: gameStart', async () => {
    const container = render(
      <Game aiLevel={mocks.aiLevel} goToMenu={mocks.goToMenu} playerSide={mocks.playerSide} />
    );
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-btn-game-start'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });

  it('renders game play PLAYER WIN', async () => {
    // jest.useFakeTimers();
    const container = render(
      <Game aiLevel={mocks.aiLevel} goToMenu={mocks.goToMenu} playerSide={mocks.playerSide} />
    );
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-setGameStatus-playing'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-setGameboard-player-win'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-handlePlayerSelectCell-1'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-set-game-result'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    // jest.runAllTimers();
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => {
      expect(screen.getByTestId('game-main')).toBeTruthy();
    });
  });

  it('renders game play CPU WIN', async () => {
    // jest.useFakeTimers();
    const container = render(
      <Game aiLevel={mocks.aiLevel} goToMenu={mocks.goToMenu} playerSide={mocks.playerSide} />
    );
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-setGameStatus-playing'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-setGameboard-cpu-win'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-handlePlayerSelectCell-1'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await act(async () => {
      fireEvent(
        screen.getByTestId('test-set-game-result'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    // jest.runAllTimers();
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => {
      expect(screen.getByTestId('game-main')).toBeTruthy();
    });
  });

  it('renders result: playAgain', async () => {
    const container = render(
      <Game aiLevel={mocks.aiLevel} goToMenu={mocks.goToMenu} playerSide={mocks.playerSide} />
    );

    await act(async () => {
      fireEvent(
        screen.getByTestId('test-btn-play-again'),
        new MouseEvent('click', {
          bubbles: true,
          cancelable: true,
        })
      );
    });
    await new Promise((resolve) => setTimeout(resolve, 0));
    await waitFor(() => {
      expect(screen.getByTestId('game-main')).toBeTruthy();
    });
  });
});
