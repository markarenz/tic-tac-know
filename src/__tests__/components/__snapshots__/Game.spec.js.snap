// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game renders game 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="game-main"
      >
        <div
          class="absolute w-full h-full left-0 top-0"
        >
          <div
            class="absolute w-full h-full left-0 top-0 w-full h-full bgPatternGame"
          />
          <div
            class="absolute w-full h-full left-0 top-0 bg-gradient-to-t from-purple-900 mix-blend-multiply"
          />
        </div>
        <div
          class="relative h-full"
        >
          <header
            class="flex justify-between items-center bg-purple-600/50 px-4 py-4 mb-8 relative z-10"
          >
            <h1
              class="text-3xl lg:text-2xl md:text-xl text-center text-white"
            >
              <b>
                Tic-Tac-Know:
              </b>
               
              <i>
                A game that Learns as you play.
              </i>
            </h1>
            <div>
              <a
                class="text-white"
                href="https://www.markmakesstuff.com"
                rel="noreferrer"
                target="_blank"
              >
                More Stuff
              </a>
            </div>
          </header>
          <div
            class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
          >
            <div
              class="mb-4"
            >
              <div
                class="grid grid-cols-2"
              >
                <div
                  class="overflow-hidden"
                >
                  <div
                    class="transition-all duration-500 opacity-0 -translate-x-16"
                    data-testid="transition-wrap"
                  >
                    <div
                      class="uppercase"
                    >
                      <b>
                        Your turn
                      </b>
                    </div>
                  </div>
                </div>
                <div
                  class="overflow-hidden"
                />
              </div>
            </div>
            <div
              class="grid grid-cols-3"
            >
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-1"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
                id="cell-2"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-3"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
                id="cell-4"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-2  "
                id="cell-5"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
                id="cell-6"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-7"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
                id="cell-8"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-9"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                  disabled=""
                >
                   
                </button>
              </div>
            </div>
            <div
              class="text-right mt-4"
            >
              Wins: 
              0
              , Losses: 
              0
              , Draws: 
              0
            </div>
          </div>
          <div
            data-testid="gamestart"
          >
            <div
              class="transition-opacity duration-500 opacity-0"
              data-testid="transition-wrap"
            >
              <div
                class="absolute left-0 top-0 w-full h-screen bg-black/60"
              />
            </div>
            <div
              class="absolute left-0 top-0 w-full h-screen flex items-center justify-center transition-all duration-500 opacity-0 -translate-x-32"
              data-testid="transition-wrap"
            >
              <div
                class="bg-purple-900 text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-lg drop-shadow-xl"
              >
                <div
                  class="grid grid-cols-4"
                >
                  <div
                    class="pr-2"
                  >
                    <div
                      class="h-full flex justify-center align-center"
                    >
                      <div>
                        <img
                          alt="heads"
                          src="heads.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="col-span-3"
                  >
                    <div
                      class="flex justify-center items-center h-full"
                    >
                      <div>
                        <h2
                          class="text-3xl text-center"
                        >
                          <i>
                            The coin toss came up 
                            heads
                            .
                          </i>
                        </h2>
                        <h3
                          class="text-xl text-center"
                        >
                          PLAYER
                           goes first this time.
                        </h3>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="text-right"
                >
                  <button
                    class="uppercase bg-transparent text-white border-2 border-white hover:bg-white hover:text-purple-900 rounded-md py-2 px-4 transition-all hover:scale-110 duration-300"
                    data-testid="gameStart-btn-start"
                  >
                    Begin
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div>
          <button
            data-testid="test-btn-game-start"
          >
            gameStart
          </button>
          <button
            data-testid="test-btn-play-again"
          >
            gameStart
          </button>
          <button
            data-testid="test-setGameStatus-playing"
          >
            gameStatusPlaying
          </button>
          <button
            data-testid="test-setGameboard-player-win"
          >
            near-win-PLAYER
          </button>
          <button
            data-testid="test-setGameboard-cpu-win"
          >
            near-win-CPU
          </button>
          <button
            data-testid="test-handlePlayerSelectCell-1"
          >
            handlePlayerSelectCell(1)
          </button>
          <button
            data-testid="test-set-game-result"
          >
            gameComplete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="game-main"
    >
      <div
        class="absolute w-full h-full left-0 top-0"
      >
        <div
          class="absolute w-full h-full left-0 top-0 w-full h-full bgPatternGame"
        />
        <div
          class="absolute w-full h-full left-0 top-0 bg-gradient-to-t from-purple-900 mix-blend-multiply"
        />
      </div>
      <div
        class="relative h-full"
      >
        <header
          class="flex justify-between items-center bg-purple-600/50 px-4 py-4 mb-8 relative z-10"
        >
          <h1
            class="text-3xl lg:text-2xl md:text-xl text-center text-white"
          >
            <b>
              Tic-Tac-Know:
            </b>
             
            <i>
              A game that Learns as you play.
            </i>
          </h1>
          <div>
            <a
              class="text-white"
              href="https://www.markmakesstuff.com"
              rel="noreferrer"
              target="_blank"
            >
              More Stuff
            </a>
          </div>
        </header>
        <div
          class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
        >
          <div
            class="mb-4"
          >
            <div
              class="grid grid-cols-2"
            >
              <div
                class="overflow-hidden"
              >
                <div
                  class="transition-all duration-500 opacity-0 -translate-x-16"
                  data-testid="transition-wrap"
                >
                  <div
                    class="uppercase"
                  >
                    <b>
                      Your turn
                    </b>
                  </div>
                </div>
              </div>
              <div
                class="overflow-hidden"
              />
            </div>
          </div>
          <div
            class="grid grid-cols-3"
          >
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-1"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
              id="cell-2"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-3"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
              id="cell-4"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-2  "
              id="cell-5"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
              id="cell-6"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-7"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
              id="cell-8"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-9"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
                disabled=""
              >
                 
              </button>
            </div>
          </div>
          <div
            class="text-right mt-4"
          >
            Wins: 
            0
            , Losses: 
            0
            , Draws: 
            0
          </div>
        </div>
        <div
          data-testid="gamestart"
        >
          <div
            class="transition-opacity duration-500 opacity-0"
            data-testid="transition-wrap"
          >
            <div
              class="absolute left-0 top-0 w-full h-screen bg-black/60"
            />
          </div>
          <div
            class="absolute left-0 top-0 w-full h-screen flex items-center justify-center transition-all duration-500 opacity-0 -translate-x-32"
            data-testid="transition-wrap"
          >
            <div
              class="bg-purple-900 text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-lg drop-shadow-xl"
            >
              <div
                class="grid grid-cols-4"
              >
                <div
                  class="pr-2"
                >
                  <div
                    class="h-full flex justify-center align-center"
                  >
                    <div>
                      <img
                        alt="heads"
                        src="heads.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="col-span-3"
                >
                  <div
                    class="flex justify-center items-center h-full"
                  >
                    <div>
                      <h2
                        class="text-3xl text-center"
                      >
                        <i>
                          The coin toss came up 
                          heads
                          .
                        </i>
                      </h2>
                      <h3
                        class="text-xl text-center"
                      >
                        PLAYER
                         goes first this time.
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-right"
              >
                <button
                  class="uppercase bg-transparent text-white border-2 border-white hover:bg-white hover:text-purple-900 rounded-md py-2 px-4 transition-all hover:scale-110 duration-300"
                  data-testid="gameStart-btn-start"
                >
                  Begin
                </button>
              </div>
            </div>
          </div>
        </div>
        <div />
      </div>
      <div>
        <button
          data-testid="test-btn-game-start"
        >
          gameStart
        </button>
        <button
          data-testid="test-btn-play-again"
        >
          gameStart
        </button>
        <button
          data-testid="test-setGameStatus-playing"
        >
          gameStatusPlaying
        </button>
        <button
          data-testid="test-setGameboard-player-win"
        >
          near-win-PLAYER
        </button>
        <button
          data-testid="test-setGameboard-cpu-win"
        >
          near-win-CPU
        </button>
        <button
          data-testid="test-handlePlayerSelectCell-1"
        >
          handlePlayerSelectCell(1)
        </button>
        <button
          data-testid="test-set-game-result"
        >
          gameComplete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Game renders result: gameStart 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        data-testid="game-main"
      >
        <div
          class="absolute w-full h-full left-0 top-0"
        >
          <div
            class="absolute w-full h-full left-0 top-0 w-full h-full bgPatternGame"
          />
          <div
            class="absolute w-full h-full left-0 top-0 bg-gradient-to-t from-purple-900 mix-blend-multiply"
          />
        </div>
        <div
          class="relative h-full"
        >
          <header
            class="flex justify-between items-center bg-purple-600/50 px-4 py-4 mb-8 relative z-10"
          >
            <h1
              class="text-3xl lg:text-2xl md:text-xl text-center text-white"
            >
              <b>
                Tic-Tac-Know:
              </b>
               
              <i>
                A game that Learns as you play.
              </i>
            </h1>
            <div>
              <a
                class="text-white"
                href="https://www.markmakesstuff.com"
                rel="noreferrer"
                target="_blank"
              >
                More Stuff
              </a>
            </div>
          </header>
          <div
            class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
          >
            <div
              class="mb-4"
            >
              <div
                class="grid grid-cols-2"
              >
                <div
                  class="overflow-hidden"
                >
                  <div
                    class="transition-all duration-500 opacity-0 -translate-x-16"
                    data-testid="transition-wrap"
                  >
                    <div
                      class="uppercase"
                    >
                      <b>
                        Your turn
                      </b>
                    </div>
                  </div>
                </div>
                <div
                  class="overflow-hidden"
                >
                  <div
                    class="transition-all duration-500 opacity-100 translate-x-0"
                    data-testid="transition-wrap"
                  >
                    <div
                      class="text-right uppercase"
                    >
                      <b>
                        CPU Turn
                      </b>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="grid grid-cols-3"
            >
              <div
                class="text-center border-gray-500 h-32  bg-white "
                id="cell-1"
              >
                <div
                  class="w-full h-full relative flex justify-center items-center"
                >
                  <div
                    class="text-3xl animate-zoomDown"
                  >
                    o
                  </div>
                </div>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
                id="cell-2"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-3"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
                id="cell-4"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-2  "
                id="cell-5"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
                id="cell-6"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-7"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
                id="cell-8"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
              <div
                class="text-center border-gray-500 h-32   "
                id="cell-9"
              >
                <button
                  class="w-full h-full"
                  data-testid="cell-btn"
                >
                   
                </button>
              </div>
            </div>
            <div
              class="text-right mt-4"
            >
              Wins: 
              0
              , Losses: 
              0
              , Draws: 
              0
            </div>
          </div>
          <div
            data-testid="gamestart"
          >
            <div
              class="transition-opacity duration-500 opacity-100"
              data-testid="transition-wrap"
            >
              <div
                class="absolute left-0 top-0 w-full h-screen bg-black/60"
              />
            </div>
            <div
              class="absolute left-0 top-0 w-full h-screen flex items-center justify-center transition-all duration-500 opacity-100 translate-x-0"
              data-testid="transition-wrap"
            >
              <div
                class="bg-purple-900 text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-lg drop-shadow-xl"
              >
                <div
                  class="grid grid-cols-4"
                >
                  <div
                    class="pr-2"
                  >
                    <div
                      class="h-full flex justify-center align-center"
                    >
                      <div>
                        <img
                          alt="heads"
                          src="heads.png"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="col-span-3"
                  >
                    <div
                      class="flex justify-center items-center h-full"
                    >
                      <div>
                        <h2
                          class="text-3xl text-center"
                        >
                          <i>
                            The coin toss came up 
                            heads
                            .
                          </i>
                        </h2>
                        <h3
                          class="text-xl text-center"
                        >
                          PLAYER
                           goes first this time.
                        </h3>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="text-right"
                >
                  <button
                    class="uppercase bg-transparent text-white border-2 border-white hover:bg-white hover:text-purple-900 rounded-md py-2 px-4 transition-all hover:scale-110 duration-300"
                    data-testid="gameStart-btn-start"
                  >
                    Begin
                  </button>
                </div>
              </div>
            </div>
          </div>
          <div />
        </div>
        <div>
          <button
            data-testid="test-btn-game-start"
          >
            gameStart
          </button>
          <button
            data-testid="test-btn-play-again"
          >
            gameStart
          </button>
          <button
            data-testid="test-setGameStatus-playing"
          >
            gameStatusPlaying
          </button>
          <button
            data-testid="test-setGameboard-player-win"
          >
            near-win-PLAYER
          </button>
          <button
            data-testid="test-setGameboard-cpu-win"
          >
            near-win-CPU
          </button>
          <button
            data-testid="test-handlePlayerSelectCell-1"
          >
            handlePlayerSelectCell(1)
          </button>
          <button
            data-testid="test-set-game-result"
          >
            gameComplete
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      data-testid="game-main"
    >
      <div
        class="absolute w-full h-full left-0 top-0"
      >
        <div
          class="absolute w-full h-full left-0 top-0 w-full h-full bgPatternGame"
        />
        <div
          class="absolute w-full h-full left-0 top-0 bg-gradient-to-t from-purple-900 mix-blend-multiply"
        />
      </div>
      <div
        class="relative h-full"
      >
        <header
          class="flex justify-between items-center bg-purple-600/50 px-4 py-4 mb-8 relative z-10"
        >
          <h1
            class="text-3xl lg:text-2xl md:text-xl text-center text-white"
          >
            <b>
              Tic-Tac-Know:
            </b>
             
            <i>
              A game that Learns as you play.
            </i>
          </h1>
          <div>
            <a
              class="text-white"
              href="https://www.markmakesstuff.com"
              rel="noreferrer"
              target="_blank"
            >
              More Stuff
            </a>
          </div>
        </header>
        <div
          class="max-w-sm mx-auto my-4 bg-white rounded-lg p-4"
        >
          <div
            class="mb-4"
          >
            <div
              class="grid grid-cols-2"
            >
              <div
                class="overflow-hidden"
              >
                <div
                  class="transition-all duration-500 opacity-0 -translate-x-16"
                  data-testid="transition-wrap"
                >
                  <div
                    class="uppercase"
                  >
                    <b>
                      Your turn
                    </b>
                  </div>
                </div>
              </div>
              <div
                class="overflow-hidden"
              >
                <div
                  class="transition-all duration-500 opacity-100 translate-x-0"
                  data-testid="transition-wrap"
                >
                  <div
                    class="text-right uppercase"
                  >
                    <b>
                      CPU Turn
                    </b>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="grid grid-cols-3"
          >
            <div
              class="text-center border-gray-500 h-32  bg-white "
              id="cell-1"
            >
              <div
                class="w-full h-full relative flex justify-center items-center"
              >
                <div
                  class="text-3xl animate-zoomDown"
                >
                  o
                </div>
              </div>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
              id="cell-2"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-3"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
              id="cell-4"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-2  "
              id="cell-5"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-t-2 border-b-2  "
              id="cell-6"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-7"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32 border-l-2 border-r-2  "
              id="cell-8"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
            <div
              class="text-center border-gray-500 h-32   "
              id="cell-9"
            >
              <button
                class="w-full h-full"
                data-testid="cell-btn"
              >
                 
              </button>
            </div>
          </div>
          <div
            class="text-right mt-4"
          >
            Wins: 
            0
            , Losses: 
            0
            , Draws: 
            0
          </div>
        </div>
        <div
          data-testid="gamestart"
        >
          <div
            class="transition-opacity duration-500 opacity-100"
            data-testid="transition-wrap"
          >
            <div
              class="absolute left-0 top-0 w-full h-screen bg-black/60"
            />
          </div>
          <div
            class="absolute left-0 top-0 w-full h-screen flex items-center justify-center transition-all duration-500 opacity-100 translate-x-0"
            data-testid="transition-wrap"
          >
            <div
              class="bg-purple-900 text-white border-2 border-purple-400 rounded-lg p-4 w-full max-w-lg drop-shadow-xl"
            >
              <div
                class="grid grid-cols-4"
              >
                <div
                  class="pr-2"
                >
                  <div
                    class="h-full flex justify-center align-center"
                  >
                    <div>
                      <img
                        alt="heads"
                        src="heads.png"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="col-span-3"
                >
                  <div
                    class="flex justify-center items-center h-full"
                  >
                    <div>
                      <h2
                        class="text-3xl text-center"
                      >
                        <i>
                          The coin toss came up 
                          heads
                          .
                        </i>
                      </h2>
                      <h3
                        class="text-xl text-center"
                      >
                        PLAYER
                         goes first this time.
                      </h3>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="text-right"
              >
                <button
                  class="uppercase bg-transparent text-white border-2 border-white hover:bg-white hover:text-purple-900 rounded-md py-2 px-4 transition-all hover:scale-110 duration-300"
                  data-testid="gameStart-btn-start"
                >
                  Begin
                </button>
              </div>
            </div>
          </div>
        </div>
        <div />
      </div>
      <div>
        <button
          data-testid="test-btn-game-start"
        >
          gameStart
        </button>
        <button
          data-testid="test-btn-play-again"
        >
          gameStart
        </button>
        <button
          data-testid="test-setGameStatus-playing"
        >
          gameStatusPlaying
        </button>
        <button
          data-testid="test-setGameboard-player-win"
        >
          near-win-PLAYER
        </button>
        <button
          data-testid="test-setGameboard-cpu-win"
        >
          near-win-CPU
        </button>
        <button
          data-testid="test-handlePlayerSelectCell-1"
        >
          handlePlayerSelectCell(1)
        </button>
        <button
          data-testid="test-set-game-result"
        >
          gameComplete
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
